### DISCUSSION TITLE
Patching files

### DISCUSSION THREAD

**Discussion by kyurom** [emukidid/swiss-gc#822] [Q&A]
Hi new to the gamecube swiss community. I had a question whenever i start up a game it applies a patch to the file. Then if i go back to that same game after turning the cube off or returning to the swiss menu it applies the patch again. Is there a way for it to apply the patch once and never again after that. So the next time I open the game it just loads it right up.

**Comment by Extrems** [comment]
It should be quicker the next around if you haven't changed any settings.

**Reply by kyurom** [reply]
It seems like it always takes about 10-15 seconds for it to apply the patches for any game that I’m launching. Is there anyway for it to just store that patch data in a folder or something. I never really change any of the settings. Is there a setting that can do that

**Reply by Extrems** [reply]
It already is. The second time around doesn't rewrite if they didn't change.

**Reply by kyurom** [reply]
Oh okay thanks for the info

**Comment by emukidid** [comment]
We could build a hash of the settings+version+device in use so if neither have changed then we blindly don't repatch and trust the patches are good. The typical use case is someone doesn't upgrade Swiss for a while and we're scanning the files needlessly.
